flake.nix

```diff
@@ -55,9 +55,14 @@
               # Don't include the `target` directory in the nix inputs.
               filter = path: type: !pkgs.lib.hasPrefix (toString ./target) path;
             };
+            # If the zlib package is not present on `aarch64-darwin`, a linker error
+            # will occur
+            buildInputs = if system == "aarch64-darwin" then [ pkgs.zlib ] else [ ];
```

Comment:
It probably doesn't hurt to add zlib as input for all platforms

Reply:
Updated the PR